nuxeo-core/nuxeo-core-api/src/test/java/org/nuxeo/ecm/core/api/impl/TestDocumentModel.java

```diff
@@ -0,0 +1,173 @@
+        super.setUp();
+        deployBundle("org.nuxeo.ecm.core.schema");
+        deployContrib("org.nuxeo.ecm.core.api.tests", "OSGI-INF/test-documentmodel-types-contrib.xml");
+    }
+
+    @Test
+    public void testSetRemovedScalarProperty() {
+        DocumentModel doc = new DocumentModelImpl("/", "doc", "File");
+        doc.setProperty("common", "size", 10L);
+        assertNull(doc.getProperty("common", "size"));
+    }
+
+    @Test
+    public void testSetRemovedScalarPropertyValue() {
+        DocumentModel doc = new DocumentModelImpl("/", "doc", "File");
+        doc.setPropertyValue("common:size", 10L);
+        assertNull(doc.getPropertyValue("common:size"));
+    }
+
+    @Test
+    public void testSetRemovedScalarProperties() {
+        DocumentModel doc = new DocumentModelImpl("/", "doc", "File");
+        doc.setProperties("common", Collections.singletonMap("size", 10L));
+        assertNull(doc.getProperties("common").get("size"));
+    }
+
+    @Test
+    public void testSetRemovedScalarPropertyWithFallbackOnScalar() {
+        DocumentModel doc = new DocumentModelImpl("/", "doc", "File");
+        doc.setProperty("deprecated", "scalar2scalar", "test scalar");
+        assertEquals("test scalar", doc.getProperty("deprecated", "scalarfallback"));
+    }
+
+    @Test
+    public void testSetRemovedScalarPropertyValueWithFallbackOnScalar() {
+        DocumentModel doc = new DocumentModelImpl("/", "doc", "File");
+        doc.setPropertyValue("deprecated:scalar2scalar", "test scalar");
+        assertEquals("test scalar", doc.getPropertyValue("deprecated:scalarfallback"));
+    }
+
+    @Test
+    public void testSetRemovedScalarPropertiesWithFallbackOnScalar() {
+        DocumentModel doc = new DocumentModelImpl("/", "doc", "File");
+        doc.setProperties("deprecated", Collections.singletonMap("scalar2scalar", "test scalar"));
+        assertEquals("test scalar", doc.getProperties("deprecated").get("scalarfallback").toString());
+    }
+
+    @Test
+    public void testSetRemovedScalarPropertyWithFallbackOnBlob() {
+        DocumentModel doc = new DocumentModelImpl("/", "doc", "File");
+        doc.setProperty("file", "filename", "test filename");
+        assertEquals("test filename", doc.getProperty("file", "content/name"));
+    }
+
+    @Test
+    public void testSetRemovedScalarPropertyValueWithFallbackOnBlob() {
+        DocumentModel doc = new DocumentModelImpl("/", "doc", "File");
+        doc.setPropertyValue("file:filename", "test filename");
+        assertEquals("test filename", doc.getPropertyValue("file:content/name"));
+    }
+
+    @Test
+    public void testSetRemovedScalarPropertiesWithFallbackOnBlob() {
+        DocumentModel doc = new DocumentModelImpl("/", "doc", "File");
+        doc.setProperties("file", Collections.singletonMap("filename", "test filename"));
+        assertEquals("test filename", ((Blob) doc.getProperties("file").get("content")).getFilename());
+    }
```

Comment:
Tests on blobs failed because we don't have any blob under _content_. 
- Leave the exception happening, as it could happened only in cases such as setting filename before the blob, so even if we handle it, filename will be erased by blob filename.
```
org.nuxeo.ecm.core.api.NuxeoException: Trying to access a member of a null object: name
    at org.nuxeo.ecm.core.api.model.impl.primitives.BlobProperty.getMemberValue(BlobProperty.java:187)
    at org.nuxeo.ecm.core.api.model.impl.primitives.BlobProperty$ScalarMemberProperty.internalGetValue(BlobProperty.java:234)
    at org.nuxeo.ecm.core.api.model.impl.AbstractProperty.setValue(AbstractProperty.java:322)
    at org.nuxeo.ecm.core.api.model.impl.AbstractProperty.setValue(AbstractProperty.java:340)
```
- Create a StringBlob at the right moment to set filename